src/components/Filters/Filters.tsx

```diff
@@ -20,23 +28,34 @@ import { Operators } from './Filters.enums';
 const generateId = ({ operator, field, condition }, index) =>
   `${operator}-${field}-${condition}-${index}`;
 
-const getDefaultCondition = (fields) =>
-  (
+const getDefaultConditionAndValue = (fields) => {
```

Comment:
```js
const getDefaultConditionAndValue = ({conditions}) => { // you don't need whole `fields` array so pass just the first item and take only conditions key from it
  const defaultCondition  = find(propEq('isDefault', true), conditions)
  
  const {
    value: defaultConditionValue, 
    component: defaultConditionComponent
  } = defaultTo(head(conditions), defaultCondition)

  const componentDefaultValue = R.path(['props', 'defaultValue'], component);
  const defaultValue = R.propOr(componentDefaultValue, 'value', componentDefaultValue);

  return { condition: defaultConditionValue, value: defaultValue };
};

Reply:
👍 Better, shorter composition. Only the initial parameters destructuring helped a lot.